org-roam-db.el

```diff
@@ -342,6 +342,10 @@ If UPDATE-P is non-nil, first remove the file in the database."
          ;; Links correctly recognized by Org Mode
          ((eq type 'link)
           (setq link element))
+         ;; Prevent self-referencing links in ROAM_REFS
+         ((and (eq type 'node-property)
+               (string-equal (org-element-property :key element) "ROAM_REFS"))
```

Comment:
@jethrokuan `string-equal` is case sensitive (per its docstring). 
Are we letting properties to be specified by lower-case? 
I use lower-case and this condition returns `nil` for "roam_refs"...

Reply:
Sure, I'll submit another PR for this.

Reply:
Great. Thank you. There are some other places that use string-equal. I'm hoping that it won't cause cascade of changes...